<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1298</width>
    <height>1081</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Volume Raycaster</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0">
     <widget class="VolumeRenderWidget" name="volumeRenderWidget"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1298</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuLoad_transfer_function">
     <property name="title">
      <string>Load transfer function</string>
     </property>
     <addaction name="actionLoadRawTff"/>
     <addaction name="actionLoadCpTff"/>
    </widget>
    <widget class="QMenu" name="menuSave_transfer_function">
     <property name="title">
      <string>Save transfer function</string>
     </property>
     <property name="icon">
      <iconset theme="document-save-as">
       <normaloff>.</normaloff>.</iconset>
     </property>
     <addaction name="actionSaveRawTff_2"/>
     <addaction name="actionSaveCpTff"/>
    </widget>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="menuLoad_transfer_function"/>
    <addaction name="menuSave_transfer_function"/>
    <addaction name="separator"/>
    <addaction name="actionLoadState"/>
    <addaction name="actionSaveState"/>
    <addaction name="separator"/>
    <addaction name="actionLoad_environment_map"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionResetCam"/>
    <addaction name="separator"/>
    <addaction name="actionSettings"/>
    <addaction name="actionShowOverlay"/>
    <addaction name="actionShowStatusBar"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionGenerateLowResVo"/>
    <addaction name="separator"/>
    <addaction name="actionScreenshot"/>
    <addaction name="actionRecord"/>
    <addaction name="actionRecordCamera"/>
    <addaction name="actionLogInteraction"/>
    <addaction name="separator"/>
    <addaction name="actionSelectOpenCL"/>
    <addaction name="actionRealoadKernel"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="dockSettings">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
     <horstretch>1</horstretch>
     <verstretch>1</verstretch>
    </sizepolicy>
   </property>
   <property name="windowTitle">
    <string>Settings</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="1" column="0" colspan="2">
      <widget class="QGroupBox" name="gbTimeSeries">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="title">
        <string>Time series</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="2" column="1">
         <widget class="QPushButton" name="pbPlay">
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset theme="media-playback-start">
            <normaloff>.</normaloff>.</iconset>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="1" colspan="3">
         <widget class="QSlider" name="sldTimeStep">
          <property name="pageStep">
           <number>1</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QCheckBox" name="chbLoop">
          <property name="text">
           <string>Loop</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QSpinBox" name="sbTimeStep"/>
        </item>
        <item row="2" column="3">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Speed [ms]</string>
          </property>
         </widget>
        </item>
        <item row="2" column="4">
         <widget class="QSpinBox" name="sbSpeed">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="singleStep">
           <number>10</number>
          </property>
          <property name="value">
           <number>100</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QComboBox" name="cbInterpolation">
       <item>
        <property name="text">
         <string>Linear</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Quad</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="2" column="0" colspan="2">
      <widget class="colorwidgets::ColorWheel" name="colorWheel" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>200</height>
        </size>
       </property>
      </widget>
     </item>
     <item row="3" column="0" colspan="2">
      <widget class="TransferFunctionWidget" name="transferFunctionEditor" native="true"/>
     </item>
     <item row="4" column="1">
      <widget class="QPushButton" name="pbResetTff">
       <property name="text">
        <string>Reset</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0" colspan="2">
      <widget class="QGroupBox" name="groupBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>250</height>
        </size>
       </property>
       <property name="title">
        <string>Rendering</string>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="11" column="2" colspan="3">
         <widget class="QCheckBox" name="chbContRendering">
          <property name="text">
           <string>Continued rendering</string>
          </property>
         </widget>
        </item>
        <item row="8" column="2" colspan="3">
         <widget class="QCheckBox" name="chbAmbientOcclusion">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Ambient occlusion</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="6" column="0" colspan="2">
         <widget class="QRadioButton" name="rbRaycast">
          <property name="text">
           <string>Ray casting</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="3" column="2" colspan="3">
         <widget class="QPushButton" name="pbBgColor">
          <property name="autoFillBackground">
           <bool>false</bool>
          </property>
          <property name="styleSheet">
           <string notr="true">QPushButton:default {
   background: yellow;
}</string>
          </property>
          <property name="text">
           <string>Background color</string>
          </property>
         </widget>
        </item>
        <item row="10" column="2" colspan="3">
         <widget class="QCheckBox" name="chbAerial">
          <property name="text">
           <string>Aerial perspective</string>
          </property>
         </widget>
        </item>
        <item row="14" column="0" colspan="2">
         <widget class="QCheckBox" name="chbBox">
          <property name="text">
           <string>Show skipped</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="13" column="2" colspan="2">
         <widget class="QCheckBox" name="chbImageESS">
          <property name="text">
           <string>Image ESS</string>
          </property>
         </widget>
        </item>
        <item row="14" column="2" colspan="3">
         <widget class="QCheckBox" name="chbGradient">
          <property name="text">
           <string>Gradient background</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="6" column="2" colspan="3">
         <widget class="QRadioButton" name="rbPathtrace">
          <property name="text">
           <string>Path tracing</string>
          </property>
         </widget>
        </item>
        <item row="11" column="0" colspan="2">
         <widget class="QCheckBox" name="chbOrtho">
          <property name="text">
           <string>Orthographic</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Ray</string>
          </property>
         </widget>
        </item>
        <item row="10" column="0" colspan="2">
         <widget class="QCheckBox" name="chbContours">
          <property name="text">
           <string>Contours</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="5">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Sampling rate</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0" colspan="2">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Shading</string>
          </property>
         </widget>
        </item>
        <item row="13" column="0" colspan="2">
         <widget class="QCheckBox" name="chbObjectESS">
          <property name="text">
           <string>Obejct ESS</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QDoubleSpinBox" name="dsbImgSampling">
          <property name="minimum">
           <double>0.010000000000000</double>
          </property>
          <property name="maximum">
           <double>8.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.010000000000000</double>
          </property>
          <property name="value">
           <double>1.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="8" column="0" colspan="2">
         <widget class="QCheckBox" name="chbLinear">
          <property name="text">
           <string>Interpolation</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="3" colspan="2">
         <widget class="QDoubleSpinBox" name="dsbSamplingRate">
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="minimum">
           <double>0.100000000000000</double>
          </property>
          <property name="maximum">
           <double>20.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>1.500000000000000</double>
          </property>
         </widget>
        </item>
        <item row="4" column="0" colspan="5">
         <widget class="QComboBox" name="cbIllum">
          <property name="currentIndex">
           <number>1</number>
          </property>
          <item>
           <property name="text">
            <string>Off</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Central differences</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Central differences &amp; transfer function</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Sobel filter</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Magnitude</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Cel shading</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Image</string>
          </property>
         </widget>
        </item>
        <item row="7" column="0" colspan="5">
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionOpen">
   <property name="icon">
    <iconset theme="document-open">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Open volume dataset...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="icon">
    <iconset theme="application-exit">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionSaveRawTff">
   <property name="text">
    <string>Save raw transfer functio...</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Settings</string>
   </property>
  </action>
  <action name="actionLoadRawTff">
   <property name="text">
    <string>From raw values...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionLoadCpTff">
   <property name="text">
    <string>From control points...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="actionSaveRawTff_2">
   <property name="text">
    <string>As raw values...</string>
   </property>
  </action>
  <action name="actionSaveCpTff">
   <property name="text">
    <string>As control points...</string>
   </property>
  </action>
  <action name="actionGenerateLowResVo">
   <property name="text">
    <string>Generate low resolution volume...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionScreenshot">
   <property name="icon">
    <iconset theme="insert-image">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Save current screen</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionRecord">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Record rendered images</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionResetCam">
   <property name="icon">
    <iconset theme="view-refresh">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Reset camera</string>
   </property>
   <property name="shortcut">
    <string>Home</string>
   </property>
  </action>
  <action name="actionSaveState">
   <property name="text">
    <string>Save state...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionLoadState">
   <property name="text">
    <string>Load state...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actionShowOverlay">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Overlay</string>
   </property>
  </action>
  <action name="actionFullscreen">
   <property name="text">
    <string>Fullscreen</string>
   </property>
   <property name="shortcut">
    <string>F11</string>
   </property>
  </action>
  <action name="actionShowStatusBar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Status bar</string>
   </property>
  </action>
  <action name="actionSelectOpenCL">
   <property name="text">
    <string>Select OpenCL platform/device...</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset theme="help-about">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>About...</string>
   </property>
  </action>
  <action name="actionRealoadKernel">
   <property name="icon">
    <iconset theme="view-refresh">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Reaload kernels</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+K</string>
   </property>
  </action>
  <action name="actionRecordCamera">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Record camera configuration</string>
   </property>
  </action>
  <action name="actionLogInteraction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Log interaction</string>
   </property>
  </action>
  <action name="actionLoad_environment_map">
   <property name="text">
    <string>Load environment map...</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>VolumeRenderWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>src/qt/volumerenderwidget.h</header>
  </customwidget>
  <customwidget>
   <class>TransferFunctionWidget</class>
   <extends>QWidget</extends>
   <header>src/qt/transferfunctionwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>colorwidgets::ColorWheel</class>
   <extends>QWidget</extends>
   <header>src/qt/colorwheel.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>dsbImgSampling</tabstop>
  <tabstop>dsbSamplingRate</tabstop>
  <tabstop>pbBgColor</tabstop>
  <tabstop>cbIllum</tabstop>
  <tabstop>sldTimeStep</tabstop>
  <tabstop>sbTimeStep</tabstop>
  <tabstop>cbInterpolation</tabstop>
  <tabstop>pbResetTff</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>465</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sldTimeStep</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sbTimeStep</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1193</x>
     <y>335</y>
    </hint>
    <hint type="destinationlabel">
     <x>1239</x>
     <y>341</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbTimeStep</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sldTimeStep</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1247</x>
     <y>329</y>
    </hint>
    <hint type="destinationlabel">
     <x>1181</x>
     <y>342</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSettings</sender>
   <signal>toggled(bool)</signal>
   <receiver>dockSettings</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>1156</x>
     <y>540</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShowStatusBar</sender>
   <signal>toggled(bool)</signal>
   <receiver>statusBar</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>648</x>
     <y>1069</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
